File: modelMain/src/main/java/com/wynne/knowledge/main/retrofit/RetrofitActivity.java
package com.wynne.knowledge.main.retrofit;

import android.util.Log;

import com.wynne.knowledge.base.base.BaseActivity;
import com.wynne.knowledge.base.model.BaseListModel;
import com.wynne.knowledge.base.model.BaseModel;
import com.wynne.knowledge.main.R;
import com.wynne.knowledge.main.model.ArticleModel;
import com.wynne.knowledge.main.net.RetrofitService;

import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;

/**
 * @author Wynne
 */
public class RetrofitActivity extends BaseActivity {
    @Override
    public void initView() {

        requestNet(RetrofitService.getInstance().obtainMains(0),
                new Observer<BaseModel<BaseListModel<ArticleModel>>>() {
            @Override
            public void onSubscribe(Disposable d) {

            }

            @Override
            public void onNext(BaseModel<BaseListModel<ArticleModel>> o) {
                for (ArticleModel articleModel : o.getData().getDatas()) {
                    Log.d("XXW", "toString: " + articleModel.getAuthor());
                }
            }

            @Override
            public void onError(Throwable e) {

            }

            @Override
            public void onComplete() {

            }
        });

    }

    @Override
    public int getLayoutId() {
        return R.layout.retrofit_activity;
    }
}

File: model_android/src/main/java/com/wynne/android/third/retrofit/RetrofitActivity.kt
package com.wynne.android.third.retrofit

import android.util.Log
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import com.wynne.android.R
import com.wynne.knowledge.base.base.BaseActivity
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitActivity : BaseActivity() {
    lateinit var disposable: Disposable

    override fun initView() {

        val build = Retrofit.Builder()
                .baseUrl("https://www.wanandroid.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build()

        val create = build.create(RetrofitService::class.java)
        disposable = create.obtainDetail()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe({
                    it.data.forEach { item ->
                        Log.d("XXW", item.name)
                    }
                }, {
                    Log.d("xxw", it.message)
                })
    }

    override fun onDestroy() {
        super.onDestroy()
        disposable.dispose()
    }


    override val layoutId: Int = R.layout.actiivty_retrofit_layout
}